### PR TITLE
Fix client keys Ownership for 3.7.x and previous versions

### PR DESCRIPTION
**Pull request by c-bordon**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-packages/issues/1025|

## Description
The ownership change process is moved so that it is the last thing to do in the upgrade process

A validation is added for the ossec-authd daemon, if it is started, the process is eliminated before the upgrade

## Tests
https://devel.ci.wazuh.info/view/Tests/job/Test_upgrade_tier/1496/consoleFull

Screenshot_20211228_081025
Screenshot_20211228_081044

- Build the package in any supported platform
  - [x] Linux
  - [ ] Windows
  - [ ] macOS
  - [ ] Solaris
  - [ ] AIX
  - [ ] HP-UX
- [x] Package installation
- [x] Package upgrade
- [ ] Package downgrade
- [ ] Package remove
- [x] Package install/remove/install
- [ ] Change added to CHANGELOG.md

- Tests for Linux RPM
  - [x] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] `%files` section is correctly updated if necessary
- Tests for Linux deb
  - [x] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] Package install/remove/install
  - [ ] Package install/purge/install
  - [ ] Check file permissions after installing the package
- Tests for macOS
  - [ ] Test the package from macOS Sierra to Mojave
- Tests for Solaris
  - [ ] Test the package on Solaris 10
  - [ ] Test the package on Solaris 11
  - [ ] Check file permissions on Solaris 11 template
- Tests for IBM AIX
  - [ ] `%files` section is correctly updated if necessary
  - [ ] Check the changes from IBM AIX 5 to 7

### CODE REVIEW

File: debs/SPECS/4.3.0/wazuh-manager/debian/preinst [hunk 1]
```diff
@@ -21,6 +21,9 @@ case "$1" in
     install|upgrade)
 
         if [ "$1" = "upgrade" ]; then
+            if pgrep -f ossec-authd > /dev/null 2>&1; then
```
Comment (alberpilot) [inline debs/SPECS/4.3.0/wazuh-manager/debian/preinst:24]: I.e: upgrade from 4.0.0 to 4.3.0, this code kills unnecessarily the `ossec-authd` daemon. This kill must be done only in those cases where none of the `wazuh` or `ossec` control stops works.

File: rpms/SPECS/4.3.0/wazuh-manager-4.3.0.spec [hunk 2]
```diff
@@ -183,6 +183,10 @@ if ! id -u wazuh > /dev/null 2>&1; then
 fi 
 
 # Stop the services to upgrade the package
+if pgrep -f ossec-authd > /dev/null 2>&1; then
```
Comment (alberpilot) [inline rpms/SPECS/4.3.0/wazuh-manager-4.3.0.spec:186]: Same

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review the requested changes

File: debs/SPECS/4.3.0/wazuh-manager/debian/preinst (hunk 1 above)
Comment (c-bordon) [reply debs/SPECS/4.3.0/wazuh-manager/debian/preinst]: Done: https://github.com/wazuh/wazuh-packages/pull/1123/commits/c8a4ee15953c804a14a9ef4b3ed827f0092ef037

File: rpms/SPECS/4.3.0/wazuh-manager-4.3.0.spec (hunk 2 above)
Comment (c-bordon) [reply rpms/SPECS/4.3.0/wazuh-manager-4.3.0.spec]: Done: https://github.com/wazuh/wazuh-packages/pull/1123/commits/c8a4ee15953c804a14a9ef4b3ed827f0092ef037

**Comment by c-bordon** [comment]
New tests were run after the changes
https://devel.ci.wazuh.info/view/Tests/job/Test_upgrade_tier/1499/console

**Review by okynos — APPROVED** [review]
LGTM.

**Review by alberpilot — APPROVED** [review]
LGTM

**Pull request merged**